### PR TITLE
Handle build failure for task support

### PR DESCRIPTION
**Pull request by BIKA-C**
### This changes report when build has failed when using task

The following changes are proposed:

- Return error code when build has failed to indicate the task has faield
- Reveal the problem panel when build has failed like the java debugger extension

### CODE REVIEW

**Comment by bobbrow** [comment]
Thank you for the PR!  We will take a look in the next few days and let you know if we have any feedback.  Please also add a short description of the change in `CHANGELOG.md` and give yourself credit for your work.

**Comment by BIKA-C** [comment]
@microsoft-github-policy-service agree company="default"

File: src/cmakeTaskProvider.ts [hunk 1]
```diff
@@ -373,10 +373,17 @@ export class CustomBuildTaskTerminal implements vscode.Pseudoterminal, proc.Outp
             const result: proc.ExecutionResult = await proc.execute(cmakePath, args, this, this.options).result;
             if (result.retc) {
                 this.writeEmitter.fire(localize("build.finished.with.error", "{0} finished with error(s).", taskName) + endOfLine);
-            } else if (result.stderr && !result.stdout) {
-                this.writeEmitter.fire(localize("build.finished.with.warnings", "{0} finished with warning(s).", taskName) + endOfLine);
-            } else if (result.stdout && result.stdout.includes("warning")) {
+                if (doCloseEmitter) {
```
Comment (bobbrow) [inline src/cmakeTaskProvider.ts:376]: This shouldn't be done here.  There is a `this.closeEmitter.fire` call a few lines down where you should pass in `result.retc ?? 0` instead.

File: src/cmakeTaskProvider.ts [hunk 2]
```diff
@@ -373,10 +373,17 @@ export class CustomBuildTaskTerminal implements vscode.Pseudoterminal, proc.Outp
...
+                    this.closeEmitter.fire(result.retc);
+                }
+                vscode.commands.executeCommand("workbench.actions.view.problems")
```
Comment (bobbrow) [inline src/cmakeTaskProvider.ts:379]: I don't think this is the correct approach.  To get the problems to focus, you need a problem matcher and then VS Code will do the right thing based on your settings in the task.  Add `"problemMatcher": "$msCompile"` to your task to make this work automatically.

File: src/cmakeTaskProvider.ts [hunk 3]
```diff
@@ -373,10 +373,17 @@ export class CustomBuildTaskTerminal implements vscode.Pseudoterminal, proc.Outp
...
+                return result.retc;
```
Comment (bobbrow) [inline src/cmakeTaskProvider.ts:380]: Don't return early here.  Just update the `return 0;` down below to be `return result.retc ?? 0;`

File: src/cmakeTaskProvider.ts [hunk 4]
```diff
@@ -373,10 +373,17 @@ export class CustomBuildTaskTerminal implements vscode.Pseudoterminal, proc.Outp
...
+                
+              // to capture warning message. Previously the condition could not capture the case if both stderr and stdout have output
+            } else if (result.stderr || (result.stdout && result.stdout.includes("warning"))) {
                 this.writeEmitter.fire(localize("build.finished.with.warnings", "{0} finished with warning(s).", taskName) + endOfLine);
+                vscode.commands.executeCommand("workbench.actions.view.problems")
```
Comment (bobbrow) [inline src/cmakeTaskProvider.ts:385]: Please revert these two lines. VS Code should do the right thing based on your task definition.

**Review by bobbrow — CHANGES_REQUESTED** [review]

**Comment by BIKA-C** [comment]
Yep yep. It's all fixed

**Review by bobbrow — APPROVED** [review]

**Pull request merged**